packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -14,28 +14,58 @@
  * limitations under the License.
  */
 
-import { diag, SpanAttributeValue, SpanAttributes } from '@opentelemetry/api';
+import { diag, AttributeValue, Attributes } from '@opentelemetry/api';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+
+const valueSanitizers: ReadonlyMap<
+  string,
+  (val: AttributeValue) => AttributeValue | undefined
+> = new Map([[SemanticAttributes.HTTP_URL, sanitizeHttpUrl]]);
+
+export function sanitizeAttribute(
+  key: string,
+  val: AttributeValue
+): AttributeValue | undefined {
+  let out: AttributeValue | undefined = undefined;
+
+  if (!isAttributeKey(key)) {
+    diag.warn(`Invalid attribute key: ${key}`);
+    return out;
+  }
+  if (!isAttributeValue(val)) {
+    diag.warn(`Invalid attribute value set for key: ${key}`);
+    return out;
+  }
+
+  let copyVal: AttributeValue;
+
+  if (Array.isArray(val)) {
+    copyVal = val.slice();
+  } else {
+    copyVal = val;
+  }
+
+  const valueSanitizer = valueSanitizers.get(key);
+  if (typeof valueSanitizer === 'function') {
+    out = valueSanitizer(copyVal);
+  } else {
+    out = copyVal;
+  }
```

Comment:
It can only be a function or undefined, right?
If so, I think it better to check whether undefined and just return, or `return valueSnitizer(copyVal)`